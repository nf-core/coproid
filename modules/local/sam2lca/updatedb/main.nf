process SAM2LCA_UPDATEDB {
    tag "${acc2tax_name}"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/sam2lca:1.1.4--pyhdfd78af_0':
        'biocontainers/sam2lca:1.1.4--pyhdfd78af_0' }"

    input:
        val(acc2tax_name)
        val(taxo_db_name)
        path(taxo_nodes)// nodes.dmp
        path(taxo_names) // names.dmp
        path(taxo_merged) // merged.dmp
        path(acc2tax_json) // optional
        path(acc2tax) // acc2tax.gz
        path(acc2tax_md5) // acc2tax.gz.md5

    output:
        path "sam2lca_db" , emit: sam2lca_db
        path "versions.yml" , emit: versions

    when:
        task.ext.when == null || task.ext.when

    script:
        def args = task.ext.args ?: ''
        def names = taxo_names ? "--taxo_names ${taxo_names}" : ''
        def nodes = taxo_nodes ? "--taxo_nodes ${taxo_nodes}" : ''
        def merged = taxo_merged ? "--taxo_merged ${taxo_merged}" : ''
        def json = acc2tax_json ? "--acc2tax_json ${acc2tax_json}" : ''
        """
        mkdir -p sam2lca_db

        sam2lca -d sam2lca_db \\
            update-db \\
            -t $taxo_db_name \\
            $names \\
            $nodes \\
            $merged \\
            -a $acc2tax_name \\
            $json \\
            $args

        cat <<-END_VERSIONS > versions.yml
        "${task.process}":
            sam2lca: \$(echo \$(sam2lca --version 2>&1) | sed 's/^sam2lca, version //' )
        END_VERSIONS
        """

    stub:
        """
        mkdir -p sam2lca_db
        touch sam2lca_db/test.pkl

        cat <<-END_VERSIONS > versions.yml
        "${task.process}":
            sam2lca: \$(echo \$(sam2lca --version 2>&1) | sed 's/^sam2lca, version //' )
        END_VERSIONS
        """
}
