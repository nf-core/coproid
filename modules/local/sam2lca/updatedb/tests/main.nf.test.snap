{
    "test-sam2lca-updatedb - test taxonomy": {
        "content": [
            {
                "0": [
                    [
                        "merged.dmp:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "merged.dmp.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "merged.dmp.gz.md5:md5,f6e01130c21a58a3371eddec53a18f6f",
                        "names.dmp:md5,e7994ec89470481e031b3ecef616e778",
                        "names.dmp.gz:md5,e7994ec89470481e031b3ecef616e778",
                        "names.dmp.gz.md5:md5,ce7546bbac7dcbe5c0054975538d7fb7",
                        "nodes.dmp:md5,9e934f98f3c2ace17fa3d77eb235f96f",
                        "nodes.dmp.gz:md5,9e934f98f3c2ace17fa3d77eb235f96f",
                        "nodes.dmp.gz.md5:md5,a4597c31842067abe62dda359f8bd854",
                        [
                            "000005.log:md5,9167e183e1dc6070dbfc81c4674d9654",
                            "CURRENT:md5,6752a1d65b201c13b62ea44016eb221f",
                            "IDENTITY:md5,730568fb30c4bd8dc6db1b95d288b9dc",
                            "LOCK:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "LOG:md5,a6e02bcd897ccd9a28a40cf3dd350292",
                            "MANIFEST-000004:md5,17211b4c15bffd11c80dfbd5b7db77ef",
                            "OPTIONS-000007:md5,22187b853e27d095e1eac121574b1d95"
                        ],
                        "test.pkl:md5,d9f5e1d08d7b678281ac088cdca355c9"
                    ]
                ],
                "1": [
                    "versions.yml:md5,175fb2cc18a30f7ab660efe1a42b7161"
                ],
                "sam2lca_db": [
                    [
                        "merged.dmp:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "merged.dmp.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "merged.dmp.gz.md5:md5,f6e01130c21a58a3371eddec53a18f6f",
                        "names.dmp:md5,e7994ec89470481e031b3ecef616e778",
                        "names.dmp.gz:md5,e7994ec89470481e031b3ecef616e778",
                        "names.dmp.gz.md5:md5,ce7546bbac7dcbe5c0054975538d7fb7",
                        "nodes.dmp:md5,9e934f98f3c2ace17fa3d77eb235f96f",
                        "nodes.dmp.gz:md5,9e934f98f3c2ace17fa3d77eb235f96f",
                        "nodes.dmp.gz.md5:md5,a4597c31842067abe62dda359f8bd854",
                        [
                            "000005.log:md5,9167e183e1dc6070dbfc81c4674d9654",
                            "CURRENT:md5,6752a1d65b201c13b62ea44016eb221f",
                            "IDENTITY:md5,730568fb30c4bd8dc6db1b95d288b9dc",
                            "LOCK:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "LOG:md5,a6e02bcd897ccd9a28a40cf3dd350292",
                            "MANIFEST-000004:md5,17211b4c15bffd11c80dfbd5b7db77ef",
                            "OPTIONS-000007:md5,22187b853e27d095e1eac121574b1d95"
                        ],
                        "test.pkl:md5,d9f5e1d08d7b678281ac088cdca355c9"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,175fb2cc18a30f7ab660efe1a42b7161"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-22T14:42:14.067314457"
    },
    "sam2lca-updatedb - stub": {
        "content": [
            {
                "0": [
                    [
                        "test.pkl:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,175fb2cc18a30f7ab660efe1a42b7161"
                ],
                "sam2lca_db": [
                    [
                        "test.pkl:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,175fb2cc18a30f7ab660efe1a42b7161"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-22T14:43:21.32148453"
    }
}